api/src/main/java/io/grpc/ServerCall.java

```diff
@@ -229,4 +229,10 @@ public String getAuthority() {
    * The {@link MethodDescriptor} for the call.
    */
   public abstract MethodDescriptor<ReqT, RespT> getMethodDescriptor();
+
+  /**
+   * Returns the {@link Server} that dispatches the call.
+   */
+  // TODO(chengyuanzhang): add ExperimentalApi annotation.
+  public abstract Server getServer();
```

Comment:
It must have a default implementation. Either return null or UnsupportedOperationException. The documentation should say what the default behavior is when unsupported.

Reply:
Default to return null.

Some thoughts about this: isn't the semantics of a `ServerCall` a call dispatched by a server? Then it's kind of reasonable to always require concrete implementations (the only real one in gRPC is `ServerCallImpl`).

Reply:
The problem is this is an API anyone can implement, and so wouldn't already be implementing this method. The "common case" would return non-null, but that doesn't mean we can guarantee all cases will.

Reply:
(And note, it isn't entirely theoretical. _Many_ interceptors implement this API. We just so happen to encourage them to use ForwardingServerCall, but it is possible they don't.)